src/utils/protectedRoute.jsx

```diff
@@ -1,11 +1,18 @@
 import { Navigate } from "react-router-dom";
 import { useAuth } from "./useAuth";
 import PropTypes from "prop-types";
+import { Spin } from "antd";
 
 //This component will be used to protect routes that require you to be signed in.
 const ProtectedRoute = ({ component: Component }) => {
-  const { isLoggedIn } = useAuth();
+  const { isLoggedIn, loading } = useAuth();
 
+  //wait for confrimLogin to finish before rendering the component
+  if (loading) {
+    return <Spin size="large" />;
+  }
```

Comment:
This is a cool change. Just out of curiosity, did you ever run into a scenario where confirmLogin was still running while you were trying to navigate to the page?

Reply:
Like have you actually been able to see this UI when running the app?

Reply:
Yeah I think if you're on the analysis page and just keep hitting the refresh browser button quickly, then you'll see it for a fraction of a second each time haha 

It could probably be centered on the page